### DISCUSSION TITLE
Is there any way in ezdxf to get the data of the bounding box based on the entity?

### DISCUSSION THREAD

**Discussion by luoyu-123** [mozman/ezdxf#323] [Q&A]
best wishes

**Comment by mozman** [comment]
Not yet, I am working on that topic, but ezdxf is not a CAD application and has no rendering engine and the data structures are not optimized for such tasks, therefore:
1. The bounding box calculation is very slow, because there is not buffered or cached geometry
2. Font measurement without font rendering is just an rough estimation, based on a mono-space font. Inline formatting, text wrapping, multi columns and many other features of MTEXT are not supported. In fact bounding box calculation for any text entity is pretty useless. Using matplotlib could improve font measurements, but this would be an optional feature, matplotlib is not a hard dependency.
3. ACIS based entities are not supported
4. XREFS are not supported

**Reply by mbway** [reply]
I wouldn't sell your efforts short though. I've found that in 99% of cases ezdxf can successfully interpret and draw the CAD files perfectly which is much more impressive than anything else I've encountered in the open source CAD world.

Applications requiring high performance can modify the approaches that ezdxf takes to achieve the same goal if they want. I would say the important thing for ezdxf is correctness and making common tasks easy.

**Reply by luoyu-123** [reply]
非常感谢！
非常感谢您提供了ezdxf这个非常完善的包
最好的祝福
